app/commands/exercise/representation/search.rb

```diff
@@ -24,8 +24,10 @@ def call
 
   private
   def filter_with_feedback!
-    if with_feedback
+    if with_feedback && mentor
       @representations = @representations.with_feedback_by(mentor)
+    elsif with_feedback && !mentor
```

Comment:
You can omit the second part, as otherwise the previous clause would have matched

```suggestion
    elsif with_feedback
```